openwisp_users/accounts/adapter.py

```diff
@@ -0,0 +1,16 @@
+from allauth.account.adapter import DefaultAccountAdapter
+from django.utils.translation import gettext as _
+from openwisp_utils.admin_theme import settings as utils_settings
+
+
+class EmailAdapter(DefaultAccountAdapter):
+    def send_mail(self, template_prefix, email, context):
+        context["logo_url"] = utils_settings.OPENWISP_EMAIL_LOGO
+        site_name = context["current_site"].name
+        context["subject"] = _("Welcome to %(site_name)s") % {"site_name": site_name}
+        context["message"] = _("Thank you for using <b>%(site_name)s</b>") % {
+            "site_name": site_name
+        }
+        context["call_to_action_url"] = context["activate_url"]
+        context["call_to_action_text"] = _("Confirm")
+        super().send_mail(template_prefix, email, context)
```

Comment:
You know what bugs me about this solution?
If we change our `send_email` and/or our templates in openwisp-utils, we may have to come back to here and update this too, which means double work for us, which is a very unpleasant solution to be in, because we'll be uncentivized to make improvements and we have to do our best to avoid this and I think it's avoidable by structuring the code differently.

I think we could try a different approach which will allow us to make improvements in the email theme feature of openwisp-utils without having to come back here to update this code.

Let's look at the `send_mail()` method of the current default adapter of allauth:

https://github.com/pennersr/django-allauth/blob/2fa7038a53e77dad504624584d2a533d08a3e7d1/allauth/account/adapter.py#L99-L136

If you take a close look, you will notice that it basically does almost the same operations we now do in openwisp-utils, except that our function is reusable across different modules, the logic of allauth only applies to allauth.

I believe we should simply rewrite the `send_mail()` method of our adapter to use the `send_email` function of openwisp-utils.

As allauth does, we should look for both the `html` and `txt` template files and render both, so we can send both text and html versions.

The difference is that in the HTML template we won't extend the base HTML email theme of openwisp-utils, but we'll generate only the message part. See my following comments for more info.